lib/js/jslib.js

```diff
@@ -4217,27 +4345,29 @@ var debugChromiumGC = _debugChromiumGC;
 function _stringEq(l, r) {
   return l === r;
 }
-var stringEq = _stringEq;
+const stringEq = _stringEq; // TODO(dhil): these definitions appear
```

Comment:
Hello,
I was the intern who wrote this nearly 8 years ago.

I am still subscribed to this repo after 8 years, partly out of sentiment -- my account was originally created to work on Links. I've been hanging around GitHub a lot recently, I saw the title of your PR and clicked out of curiosity.

I think I can provide some context.

All the functions starting here:
https://github.com/links-lang/links/blob/master/core/lib.ml#L1420

`(* LINKS GAME LIBRARY *)`

up to here:

https://github.com/links-lang/links/blob/master/core/lib.ml#L1555

`(* END OF EQUALITY FUNCTIONS *)`

were written for the purpose of implementing these example games:

https://github.com/links-lang/links/tree/master/examples/games

I suspect they haven't been used elsewhere, as I don't think they were documented.

The idea was to make the games run fast enough to be playable. Optimizations were necessary to do that.

They were achieved with by any means I could come up with, not all of which were necessarily pretty or sensible as I might've not known any better -- I was far from competent on anything I was doing. For example in this definition of `_objectEq`:

```js
function _objectEq(l, r) {
  return l._label === r._label && (l._value === r._value || (l._value === {} && r._value === {}));
}
```

I'm not sure what I was thinking when I wrote `(l._value === {} && r._value === {})`. It seems to be useless nonsense (fortunately harmless).

Anycase I think these functions can be safely refactored as long as it is verified that the game examples run sensibly fast.

Hope this helps.

Cheers,
Dariusz

PS I'd create a PR, but I couldn't get past the setup. I hope this is better than nothing.

Reply:
Hi Dariusz! 

Thanks for the clarification. In this patch, I have updated and implemented all the infix operators on the client side. The performance difference between my definitions and those used by the games should be negligible. The JS engines have become really apt at optimising code. 

I think my comment about the brokenness of those functions refer to the fact that they drop the current continuation, meaning they cannot be used in an effectful or higher-order context. Though, they were probably OK because they were marked `PURE` in `lib.ml`, so the JavaScript compiler should eta-expand them accordingly.

Ideally, I think we want the JavaScript compiler to be type-directed such that it can insert the efficient native versions of equality, relational, string operators.